pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -561,30 +595,86 @@ func (sip *streamIngestionProcessor) bufferSST(sst *roachpb.RangeFeedSSTable) er
 	_, sp := tracing.ChildSpan(sip.Ctx, "stream-ingestion-buffer-sst")
 	defer sp.Finish()
 
-	iter, err := storage.NewMemSSTIterator(sst.Data, true)
+	iter, err := storage.NewPebbleMemSSTIterator(sst.Data, true,
+		storage.IterOptions{
+			KeyTypes: storage.IterKeyTypePointsAndRanges,
+		})
+
 	if err != nil {
 		return err
 	}
 	defer iter.Close()
 	for ; ; iter.Next() {
-		if ok, err := iter.Valid(); err != nil {
+		if valid, err := iter.Valid(); err != nil {
 			return err
-		} else if !ok { // cursor passed the span end key
+		} else if !valid { // cursor passed the span end key
 			break
 		}
-		if err = sip.bufferKV(&roachpb.KeyValue{
-			Key: iter.UnsafeKey().Key,
-			Value: roachpb.Value{
-				RawBytes:  iter.UnsafeValue(),
-				Timestamp: iter.UnsafeKey().Timestamp,
-			},
-		}); err != nil {
-			return err
+		if hasPoint, hasRange := iter.HasPointAndRange(); hasPoint {
+			if err = sip.bufferKV(&roachpb.KeyValue{
+				Key: iter.UnsafeKey().Key,
+				Value: roachpb.Value{
+					RawBytes:  iter.UnsafeValue(),
+					Timestamp: iter.UnsafeKey().Timestamp,
+				},
+			}); err != nil {
+				return err
+			}
+		} else if hasRange {
```

Comment:
This isn't strictly correct, since it's possible for a range key and point key to be initially surfaced together. This can only happen for point keys without timestamps, i.e. intents or inline values, where the point key and range key's start bound are identical. Normally, we handle this by caching and comparing `RangeBounds().Key` (until #84379).

However, we can do something far more efficient here, by processing point and range keys separately, since we ingest them separately anyway. That is, first use `IterKeyTypePointsOnly` to process the point keys, then `IterKeyTypeRangesOnly` to process the range keys.